lib/elixir/lib/kernel.ex

```diff
@@ -3635,6 +3635,16 @@ defmodule Kernel do
   defp collect_traces(_name, arg, %{lexical_tracker: pid} = env) when is_pid(pid) do
     env = %{env | function: {:__info__, 1}}
 
+    arg =
+      case arg do
+        {{:., _, [{:__aliases__, _, [:Application]}, :compile_env]}, _, [app, key, default]} ->
+          default = Macro.expand_literals(default, env)
+          put_elem(arg, 2, [app, key, default])
+
+        arg ->
+          arg
+      end
```

Comment:
To be clear, this should be done inside `Macro.expand_literals/2` itself! For example, you could add this test instead:

```
@example %{foo: Application.compile_env(:elixir, __MODULE__, String)}
```

This will make it so this issue and the previous issue you reported "compose". :)

Reply:
I had previously put it in Kernel before seeing your earlier comment. Now I pushed a commit that moves this logic to Macro